sway-lsp/src/utils/keyword_docs.rs

```diff
@@ -0,0 +1,828 @@
+            mod false_keyword {}
+        };
+
+        let break_keyword: ItemMod = parse_quote! {
+            /// Exit early from a loop.
+            ///
+            /// When `break` is encountered, execution of the associated loop body is
+            /// immediately terminated.
+            ///
+            /// ```sway
+            /// let mut x = 0;
+            ///
+            /// for x < 100 {
+            ///     if x > 12 {
+            ///         break;
+            ///     }
+            ///     x += 1;
+            /// }
+            ///
+            /// assert(x == 12);
+            /// ```
+            mod break_keyword {}
+        };
+
+        let continue_keyword: ItemMod = parse_quote! {
+            /// Skip to the next iteration of a loop.
+            ///
+            /// When `continue` is encountered, the current iteration is terminated, returning control to the
+            /// loop head, typically continuing with the next iteration.
+            ///
+            /// ```sway
+            /// // Printing odd numbers by skipping even ones
+            /// for number in 1..=10 {
+            ///     if number % 2 == 0 {
+            ///         continue;
+            ///     }
+            ///     log(number);
+            /// }
+            /// ```
+            mod continue_keyword {}
+        };
+
+        // TODO
+        let for_keyword: ItemMod = parse_quote! {
+            mod for_keyword {}
+        };
+
+        // TODO
+        let where_keyword: ItemMod = parse_quote! {
+            mod where_keyword {}
+        };
+
+        // TODO
+        let ref_keyword: ItemMod = parse_quote! {
+            mod ref_keyword {}
+        };
+
+        // TODO
+        let script_keyword: ItemMod = parse_quote! {
+            mod script_keyword {}
+        };
+
+        // TODO
+        let contract_keyword: ItemMod = parse_quote! {
+            mod contract_keyword {}
+        };
+
+        // TODO
+        let predicate_keyword: ItemMod = parse_quote! {
+            mod predicate_keyword {}
+        };
+
+        // TODO
+        let library_keyword: ItemMod = parse_quote! {
+            mod library_keyword {}
+        };
+
+        // TODO
+        let dep_keyword: ItemMod = parse_quote! {
+            mod dep_keyword {}
+        };
+
+        // TODO
+        let abi_keyword: ItemMod = parse_quote! {
+            mod abi_keyword {}
+        };
+
+        // TODO
+        let storage_keyword: ItemMod = parse_quote! {
+            mod storage_keyword {}
+        };
+
+        // TODO
+        let asm_keyword: ItemMod = parse_quote! {
+            mod asm_keyword {}
+        };
+
+        // TODO
+        let deref_keyword: ItemMod = parse_quote! {
+            mod deref_keyword {}
+        };
+
+        // TODO
+        let configurable_keyword: ItemMod = parse_quote! {
+            mod configurable_keyword {}
+        };
+
+        let mut keyword_docs = HashMap::new();
+
+        let keywords = vec![
```

Comment:
It would be cool if there was a way to ensure that this list of keywords matches the keywords defined in `keywords.rs`, so we either get a compilation error or a test failure when a new keyword is added without docs.

Reply:
Maybe it's not possible with the way macros work...

Reply:
great call, i'll add a test to ensure this is always the case.